apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -207,7 +223,7 @@ export class UpsertWorkflowUseCase {
 
     return {
       id: existingWorkflow._id,
-      environmentId: user.environmentId,
+      environmentId: existingWorkflow._environmentId,
```

Comment:
This needs to be the environment where the workflow that we are updating is stored, not the user current environment.

Reply:
good point, besides the tests i would add a const with a comment to prevent future maintainers break it while refactoring the code. 
not a deal breaker just as a thought for your consideration.